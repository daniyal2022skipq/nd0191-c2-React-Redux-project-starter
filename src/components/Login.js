import "bootstrap/dist/css/bootstrap.min.css";
import { useState } from "react";
import { useSelector, useDispatch } from "react-redux";
import {Navigate} from "react-router-dom";
import { login } from "../slices/authSlice";
function Login(props) {
  const [nameError, setNameError] = useState("");
  const [passError, setPassError] = useState("");
  const [name, setName] = useState();
  const [password, setPassword] = useState();
const authUser=useSelector(state=>state.auth.authUser)
const dispatch = useDispatch();
const usersList = useSelector((state) => state.user.users);
if (authUser) {
    const urlParams = new URLSearchParams(window.location.search);
    const redirectUrl = urlParams.get('redirectTo');
    return <Navigate to={redirectUrl ? redirectUrl : "/dashboard"}/>;
  }
  const handleSubmit = (e) => {
    e.preventDefault();
    if (usersList[name] === undefined) {
      setNameError("Incorrect Name");
      setPassError("");
      return;
    }
    if (usersList[name].password !== password) {
      setPassError("Incorrect Password");
      setNameError("");
      return;
    }
    if (usersList[name].password === password) {
      dispatch(login(name));
    }
  };


  return (
    <div className="Auth-form-container">
      <form className="Auth-form" onSubmit={handleSubmit}>
        <div className="Auth-form-content">
          <img
            className="login-Img"
            src="https://img.freepik.com/free-vector/access-control-system-abstract-concept_335657-3180.jpg?t=st=1657876432~exp=1657877032~hmac=7ec8670dc6a18ca1d8961eeb0ecb1d6f132c026fd7f8691d707d15440c3d281f&w=826"
            alt='userimage'
          />
          <div className="form-group mt-3">
            <label>Username</label>
            <input
              type="text"
              className="form-control mt-1"
              placeholder="Enter Username"
              value={name}
              onChange={(e) => setName(e.target.value)}
            />
            <div data-testid="nameError">{nameError}</div>
          </div>
          <div className="form-group mt-3">
            <label>Password</label>
            <input
              type="password"
              className="form-control mt-1"
              placeholder="Enter password"
              value={password}
              onChange={(e) => setPassword(e.target.value)}
            />
            <div data-testid="passError">{passError}</div>
          </div>
          <div className="d-grid gap-2 mt-3">
            <button
              type="submit"
              className="btn btn-primary"
              disabled={!name || !password}
              data-testid="btn"
            >
              Login
            </button>
          </div>
        </div>
      </form>
    </div>
  );
}

export default Login;
