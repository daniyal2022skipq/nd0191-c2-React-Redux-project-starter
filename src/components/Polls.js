import { useDispatch, useSelector } from "react-redux";
import SinglePoll from "./SinglePoll";
import { useEffect } from "react";
import { getquestionThunk } from "../slices/questionSlice";
const Polls = () => {
  const dispatch = useDispatch();
  const questions = useSelector((state) => state.question.allQuestions);
  const authUser = useSelector((state) => state.auth.authUser);
  const users = useSelector((state) => state.user.users);
  const newQ = Object.keys(questions)
    .map((user) => {
      return {
        image: users[questions[user].author].avatarURL,
        timestamp: questions[user].timestamp,
        author: questions[user].author,
        id: questions[user].id,
        key: questions[user].id,
      };
    })
    .sort((a, b) => {
      return b.timestamp - a.timestamp;
    });
const questionLength=Object.keys(questions).length
  useEffect(() => {
    let mounted = true;
    if (mounted) {
      dispatch(getquestionThunk());
    }
    return () => {
      mounted = false;
    };
  }, [questionLength,dispatch]);
  return (
    <div>
      <div className="allPolls">
        <h2>New</h2>
        <div className="newDiv">
          {newQ.map((question) => {
            if (!users[authUser]?.answers[question.id]) {
              return (
                <SinglePoll
                  image={users[question.author].avatarURL}
                  date={question.timestamp}
                  author={question.author}
                  id={question.id}
                  key={question.id}
                />
              );
            }
            return null;
          })}
        </div>
      </div>
      <div>
        <h2>Done</h2>

        <div className="oldDiv">
          {newQ.map((question) => {
            if (users[authUser]?.answers[question.id]) {
              return (
                <SinglePoll
                  image={users[question.author].avatarURL}
                  date={question.timestamp}
                  author={question.author}
                  id={question.id}
                  key={question.id}
                />
              );
            }
            return null;
          })}
        </div>
      </div>
    </div>
  );
};

export default Polls;
