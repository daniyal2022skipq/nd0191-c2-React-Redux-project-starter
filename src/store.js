import { configureStore } from "@reduxjs/toolkit";
import authReducer from "./slices/authSlice";
import userReducer from './slices/userSlice'
import questionReducer from './slices/questionSlice'
import answerReducer from './slices/answerSlice'

const store=configureStore(
   {
    reducer:{
        auth:authReducer,
        user:userReducer,
        question:questionReducer,
        answer:answerReducer
    }
   }
)

export default store