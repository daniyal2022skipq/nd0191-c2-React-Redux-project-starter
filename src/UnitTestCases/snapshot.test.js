import {BrowserRouter} from 'react-router-dom'
import {Provider} from 'react-redux'
import Login from '../components/Login'
import {render} from '@testing-library/react';
import store from '../store'
import Polls from '../components/Polls';
test('snapshot test', () => {
    const component=render(
        <BrowserRouter>
        <Provider store={store}>
            <Polls/>
            </Provider>
            </BrowserRouter>
    )
    expect(component).toMatchSnapshot()
  })