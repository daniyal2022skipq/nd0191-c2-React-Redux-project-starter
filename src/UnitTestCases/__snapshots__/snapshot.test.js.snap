// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="allPolls"
        >
          <h2>
            New
          </h2>
          <div
            class="newDiv"
          >
            <div
              class="card"
              style="width: 10rem;"
            >
              <img
                class="card-img-top"
                src="https://cdn-icons-png.flaticon.com/512/906/906517.png?w=826&t=st=1657799200~exp=1657799800~hmac=20bd50505bd74d5b031492d5ec069dde6ccab1df5dc68780fff92f18e077fb39"
              />
              <div
                class="card-body"
              >
                <div
                  class="card-title h5"
                >
                  mtsamis
                </div>
                <p
                  class="card-text"
                >
                  12:16AM | 5/1/2017
                </p>
                <button
                  class="btn btn-primary"
                  type="button"
                >
                  View Poll
                </button>
              </div>
            </div>
            <div
              class="card"
              style="width: 10rem;"
            >
              <img
                class="card-img-top"
                src="https://img.freepik.com/free-vector/mysterious-mafia-man-smoking-cigarette_52683-34828.jpg?w=826&t=st=1657799167~exp=1657799767~hmac=c1cbdd88932e07e6fcc74ea2b439f3753450185c273062f3647a9e37472d2579"
              />
              <div
                class="card-body"
              >
                <div
                  class="card-title h5"
                >
                  tylermcginnis
                </div>
                <p
                  class="card-text"
                >
                  5:09:PM | 3/15/2017
                </p>
                <button
                  class="btn btn-primary"
                  type="button"
                >
                  View Poll
                </button>
              </div>
            </div>
            <div
              class="card"
              style="width: 10rem;"
            >
              <img
                class="card-img-top"
                src="https://img.freepik.com/free-vector/businessman-character-avatar-isolated_24877-60111.jpg?t=st=1657795176~exp=1657795776~hmac=6042db2b97f05e44e5e14a3ab210cd1685d929777cf37a3a4aa25fcc3cbacf33&w=826"
              />
              <div
                class="card-body"
              >
                <div
                  class="card-title h5"
                >
                  sarahedo
                </div>
                <p
                  class="card-text"
                >
                  3:22:AM | 3/4/2017
                </p>
                <button
                  class="btn btn-primary"
                  type="button"
                >
                  View Poll
                </button>
              </div>
            </div>
            <div
              class="card"
              style="width: 10rem;"
            >
              <img
                class="card-img-top"
                src="https://img.freepik.com/free-vector/mysterious-mafia-man-smoking-cigarette_52683-34828.jpg?w=826&t=st=1657799167~exp=1657799767~hmac=c1cbdd88932e07e6fcc74ea2b439f3753450185c273062f3647a9e37472d2579"
              />
              <div
                class="card-body"
              >
                <div
                  class="card-title h5"
                >
                  tylermcginnis
                </div>
                <p
                  class="card-text"
                >
                  4:42:PM | 12/24/2016
                </p>
                <button
                  class="btn btn-primary"
                  type="button"
                >
                  View Poll
                </button>
              </div>
            </div>
            <div
              class="card"
              style="width: 10rem;"
            >
              <img
                class="card-img-top"
                src="https://cdn-icons-png.flaticon.com/512/906/906517.png?w=826&t=st=1657799200~exp=1657799800~hmac=20bd50505bd74d5b031492d5ec069dde6ccab1df5dc68780fff92f18e077fb39"
              />
              <div
                class="card-body"
              >
                <div
                  class="card-title h5"
                >
                  mtsamis
                </div>
                <p
                  class="card-text"
                >
                  12:02PM | 7/14/2016
                </p>
                <button
                  class="btn btn-primary"
                  type="button"
                >
                  View Poll
                </button>
              </div>
            </div>
            <div
              class="card"
              style="width: 10rem;"
            >
              <img
                class="card-img-top"
                src="https://img.freepik.com/free-vector/businessman-character-avatar-isolated_24877-60111.jpg?t=st=1657795176~exp=1657795776~hmac=6042db2b97f05e44e5e14a3ab210cd1685d929777cf37a3a4aa25fcc3cbacf33&w=826"
              />
              <div
                class="card-body"
              >
                <div
                  class="card-title h5"
                >
                  sarahedo
                </div>
                <p
                  class="card-text"
                >
                  7:21:AM | 6/29/2016
                </p>
                <button
                  class="btn btn-primary"
                  type="button"
                >
                  View Poll
                </button>
              </div>
            </div>
          </div>
        </div>
        <div>
          <h2>
            Done
          </h2>
          <div
            class="oldDiv"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="allPolls"
      >
        <h2>
          New
        </h2>
        <div
          class="newDiv"
        >
          <div
            class="card"
            style="width: 10rem;"
          >
            <img
              class="card-img-top"
              src="https://cdn-icons-png.flaticon.com/512/906/906517.png?w=826&t=st=1657799200~exp=1657799800~hmac=20bd50505bd74d5b031492d5ec069dde6ccab1df5dc68780fff92f18e077fb39"
            />
            <div
              class="card-body"
            >
              <div
                class="card-title h5"
              >
                mtsamis
              </div>
              <p
                class="card-text"
              >
                12:16AM | 5/1/2017
              </p>
              <button
                class="btn btn-primary"
                type="button"
              >
                View Poll
              </button>
            </div>
          </div>
          <div
            class="card"
            style="width: 10rem;"
          >
            <img
              class="card-img-top"
              src="https://img.freepik.com/free-vector/mysterious-mafia-man-smoking-cigarette_52683-34828.jpg?w=826&t=st=1657799167~exp=1657799767~hmac=c1cbdd88932e07e6fcc74ea2b439f3753450185c273062f3647a9e37472d2579"
            />
            <div
              class="card-body"
            >
              <div
                class="card-title h5"
              >
                tylermcginnis
              </div>
              <p
                class="card-text"
              >
                5:09:PM | 3/15/2017
              </p>
              <button
                class="btn btn-primary"
                type="button"
              >
                View Poll
              </button>
            </div>
          </div>
          <div
            class="card"
            style="width: 10rem;"
          >
            <img
              class="card-img-top"
              src="https://img.freepik.com/free-vector/businessman-character-avatar-isolated_24877-60111.jpg?t=st=1657795176~exp=1657795776~hmac=6042db2b97f05e44e5e14a3ab210cd1685d929777cf37a3a4aa25fcc3cbacf33&w=826"
            />
            <div
              class="card-body"
            >
              <div
                class="card-title h5"
              >
                sarahedo
              </div>
              <p
                class="card-text"
              >
                3:22:AM | 3/4/2017
              </p>
              <button
                class="btn btn-primary"
                type="button"
              >
                View Poll
              </button>
            </div>
          </div>
          <div
            class="card"
            style="width: 10rem;"
          >
            <img
              class="card-img-top"
              src="https://img.freepik.com/free-vector/mysterious-mafia-man-smoking-cigarette_52683-34828.jpg?w=826&t=st=1657799167~exp=1657799767~hmac=c1cbdd88932e07e6fcc74ea2b439f3753450185c273062f3647a9e37472d2579"
            />
            <div
              class="card-body"
            >
              <div
                class="card-title h5"
              >
                tylermcginnis
              </div>
              <p
                class="card-text"
              >
                4:42:PM | 12/24/2016
              </p>
              <button
                class="btn btn-primary"
                type="button"
              >
                View Poll
              </button>
            </div>
          </div>
          <div
            class="card"
            style="width: 10rem;"
          >
            <img
              class="card-img-top"
              src="https://cdn-icons-png.flaticon.com/512/906/906517.png?w=826&t=st=1657799200~exp=1657799800~hmac=20bd50505bd74d5b031492d5ec069dde6ccab1df5dc68780fff92f18e077fb39"
            />
            <div
              class="card-body"
            >
              <div
                class="card-title h5"
              >
                mtsamis
              </div>
              <p
                class="card-text"
              >
                12:02PM | 7/14/2016
              </p>
              <button
                class="btn btn-primary"
                type="button"
              >
                View Poll
              </button>
            </div>
          </div>
          <div
            class="card"
            style="width: 10rem;"
          >
            <img
              class="card-img-top"
              src="https://img.freepik.com/free-vector/businessman-character-avatar-isolated_24877-60111.jpg?t=st=1657795176~exp=1657795776~hmac=6042db2b97f05e44e5e14a3ab210cd1685d929777cf37a3a4aa25fcc3cbacf33&w=826"
            />
            <div
              class="card-body"
            >
              <div
                class="card-title h5"
              >
                sarahedo
              </div>
              <p
                class="card-text"
              >
                7:21:AM | 6/29/2016
              </p>
              <button
                class="btn btn-primary"
                type="button"
              >
                View Poll
              </button>
            </div>
          </div>
        </div>
      </div>
      <div>
        <h2>
          Done
        </h2>
        <div
          class="oldDiv"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
